src/creatures/players/player.cpp

```diff
@@ -7749,6 +7750,128 @@ void Player::closeAllExternalContainers() {
 	}
 }
 
+bool Player::canBuyStoreOffer(const Offer* offer) {
+	auto offerType = offer->getOfferType();
+	auto canBuy = true;
+
+	switch (offerType) {
+		case OfferTypes_t::OUTFIT: {
+			auto offerOutfitId = offer->getOutfitIds();
+			auto playerLookType = (getSex() == PLAYERSEX_FEMALE ? offerOutfitId.femaleId : offerOutfitId.maleId);
+			auto addons = playerLookType >= 962 && playerLookType <= 975 ? 0 : 3;
+
+			if (canWear(playerLookType, addons)) {
+				canBuy = false;
+			}
+			break;
+		}
+
+		case OfferTypes_t::MOUNT: {
+			auto mount = g_game().mounts.getMountByID(offer->getOfferId());
+
+			if (hasMount(mount)) {
+				canBuy = false;
+			}
+			break;
+		}
+
+		case OfferTypes_t::EXPBOOST: {
+			auto expBoostCount = getStorageValue(STORAGEVALUE_EXPBOOST);
+
+			if (expBoostCount >= 6 || getXpBoostTime() > 0) {
+				canBuy = false;
+			}
+			break;
+		}
+
+		case OfferTypes_t::PREYSLOT: {
+			const auto &thirdSlot = getPreySlotById(PreySlot_Three);
+
+			if (thirdSlot->state != PreyDataState_Locked) {
+				canBuy = false;
+			}
+
+			break;
+		}
+
+		case OfferTypes_t::PREYBONUS: {
+			auto cardsAmount = offer->getOfferCount();
+			if (getPreyCards() + cardsAmount >= 50) {
```

Comment:
It might be beneficial to establish a new key in the config.lua file for this purpose, thereby eliminating the magic number and enhancing organizational control.